lib/IRGen/GenMeta.cpp

```diff
@@ -3349,6 +3350,12 @@ namespace {
         flags |= ClassFlags::UsesSwift1Refcounting;
       }
 
+      DeclAttributes attrs = Target->getAttrs();
+      if (auto objc = attrs.getAttribute<ObjCAttr>()) {
```

Comment:
Ah, please check for ObjCRuntimeNameAttr as well.

(Alternately, you could make this an optional second parameter to ClassDecl::getObjCRuntimeName, so that they stay in sync.)